pulser-core/pulser/channels/base_channel.py

```diff
@@ -263,7 +271,8 @@ def Local(
             min_avg_amp: The minimum average amplitude of a pulse (when not
                 zero).
         """
-        if cls._has_fixed_addressing:
+        print(get_type_hints(cls)["addressing"])
+        if len(get_args(get_type_hints(cls)["addressing"])) == 1:
```

Comment:
This is a smart solution, but I'm not sure relying so much on type hints is such a good idea... 
Did you consider using the dataclass fields instead? You could check for `Field.init` being `False` and `default` not being `MISSING`.